### PR TITLE
Rebuild libopenvino dev2024 0 1 h50e462

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
Replacement for https://github.com/conda-forge/opencv-feedstock/pull/397
Ported PR from OpenCV upstream https://github.com/opencv/opencv/pull/25199

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
```
Encountered problems while solving:
  - package libopenvino-dev-2024.0.0-hd5fc58b_0 requires libopenvino-tensorflow-frontend 2024.0.0 he1e0747_0, but none of the providers can be installed
```

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
sorry, lets remove the dual protobuf builds, i think it caused some uploads to fail. I'll make the PR to openvino now

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
The "real" failure seems to be:
```
[428/1322] Building CXX object modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/mvn_layer.cpp.o
[429/1322] Building CXX object modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/nary_eltwise_layers.cpp.o
FAILED: modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/nary_eltwise_layers.cpp.o 
$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-c++ -DCVAPI_EXPORTS -DENABLE_PLUGINS -DHAVE_DNN_NGRAPH -DHAVE_FLATBUFFERS=1 -DHAVE_INF_ENGINE -DHAVE_NGRAPH -DHAVE_PROTOBUF=1 -DIE_THREAD=IE_THREAD_TBB -DINF_ENGINE_RELEASE=2024000000 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -DOV_THREAD=OV_THREAD_TBB -DPROTOBUF_USE_DLLS -DTBB_PREVIEW_WAITING_FOR_WORKERS=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$SRC_DIR/build/3rdparty/ippicv/ippicv_lnx/icv/include -I$SRC_DIR/build/3rdparty/ippicv/ippicv_lnx/iw/include -I$SRC_DIR/build -I$SRC_DIR/modules/dnn/include -I$SRC_DIR/build/modules/dnn -I$SRC_DIR/modules/core/include -I$SRC_DIR/modules/imgproc/include -I$SRC_DIR/modules/dnn/misc/tflite -isystem $PREFIX/include/eigen3 -isystem $SRC_DIR/3rdparty/flatbuffers/include -fvisibility-inlines-hidden -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/libopencv-4.9.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -D__STDC_CONSTANT_MACROS   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Wundef -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fopenmp -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-invalid-offsetof -O3 -DNDEBUG  -DNDEBUG -std=c++17 -fPIC -Wno-error=deprecated-declarations -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/nary_eltwise_layers.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/nary_eltwise_layers.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/nary_eltwise_layers.cpp.o -c $SRC_DIR/modules/dnn/src/layers/nary_eltwise_layers.cpp
$SRC_DIR/modules/dnn/src/layers/nary_eltwise_layers.cpp: In member function 'virtual cv::Ptr<cv::dnn::dnn4_v20231225::BackendNode> cv::dnn::NaryEltwiseLayerImpl::initNgraph(const std::vector<cv::Ptr<cv::dnn::dnn4_v20231225::BackendWrapper> >&, const std::vector<cv::Ptr<cv::dnn::dnn4_v20231225::BackendNode> >&)':
$SRC_DIR/modules/dnn/src/layers/nary_eltwise_layers.cpp:934:35: error: 'FMOD' is not a member of 'cv::dnn::NaryEltwiseLayerImpl::OPERATION'; did you mean 'MOD'?
  934 |         else if (op == OPERATION::FMOD)
      |                                   ^~~~
      |                                   MOD
[430/1322] Building CXX object modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/normalize_bbox_layer.cpp.o
```

**Comment by h-vetinari** [comment]
AFAICT, `OPERATION::FMOD` got added in https://github.com/opencv/opencv/commit/c955564cb3f8b28915a3ebe7e5f04d9fb78926bb, which hasn't been in a release yet. Not sure if it's possible to backport that commit as well.

**Comment by ilya-lavrenov** [comment]
> The "real" failure seems to be:
> 
> ```
> [428/1322] Building CXX object modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/mvn_layer.cpp.o
> [429/1322] Building CXX object modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/nary_eltwise_layers.cpp.o
> FAILED: modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/nary_eltwise_layers.cpp.o 
> $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-c++ -DCVAPI_EXPORTS -DENABLE_PLUGINS -DHAVE_DNN_NGRAPH -DHAVE_FLATBUFFERS=1 -DHAVE_INF_ENGINE -DHAVE_NGRAPH -DHAVE_PROTOBUF=1 -DIE_THREAD=IE_THREAD_TBB -DINF_ENGINE_RELEASE=2024000000 -DOPENCV_DNN_EXTERNAL_PROTOBUF=1 -DOV_THREAD=OV_THREAD_TBB -DPROTOBUF_USE_DLLS -DTBB_PREVIEW_WAITING_FOR_WORKERS=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$SRC_DIR/build/3rdparty/ippicv/ippicv_lnx/icv/include -I$SRC_DIR/build/3rdparty/ippicv/ippicv_lnx/iw/include -I$SRC_DIR/build -I$SRC_DIR/modules/dnn/include -I$SRC_DIR/build/modules/dnn -I$SRC_DIR/modules/core/include -I$SRC_DIR/modules/imgproc/include -I$SRC_DIR/modules/dnn/misc/tflite -isystem $PREFIX/include/eigen3 -isystem $SRC_DIR/3rdparty/flatbuffers/include -fvisibility-inlines-hidden -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/libopencv-4.9.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -D__STDC_CONSTANT_MACROS   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Wundef -Winit-self -Wpointer-arith -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fopenmp -Wno-deprecated -Wno-missing-declarations -Wno-shadow -Wno-unused-parameter -Wno-sign-compare -Wno-invalid-offsetof -O3 -DNDEBUG  -DNDEBUG -std=c++17 -fPIC -Wno-error=deprecated-declarations -MD -MT modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/nary_eltwise_layers.cpp.o -MF modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/nary_eltwise_layers.cpp.o.d -o modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/nary_eltwise_layers.cpp.o -c $SRC_DIR/modules/dnn/src/layers/nary_eltwise_layers.cpp
> $SRC_DIR/modules/dnn/src/layers/nary_eltwise_layers.cpp: In member function 'virtual cv::Ptr<cv::dnn::dnn4_v20231225::BackendNode> cv::dnn::NaryEltwiseLayerImpl::initNgraph(const std::vector<cv::Ptr<cv::dnn::dnn4_v20231225::BackendWrapper> >&, const std::vector<cv::Ptr<cv::dnn::dnn4_v20231225::BackendNode> >&)':
> $SRC_DIR/modules/dnn/src/layers/nary_eltwise_layers.cpp:934:35: error: 'FMOD' is not a member of 'cv::dnn::NaryEltwiseLayerImpl::OPERATION'; did you mean 'MOD'?
>   934 |         else if (op == OPERATION::FMOD)
>       |                                   ^~~~
>       |                                   MOD
> [430/1322] Building CXX object modules/dnn/CMakeFiles/opencv_dnn.dir/src/layers/normalize_bbox_layer.cpp.o
> ```

Great catch! Updated the patch

**Comment by hmaarrfk** [comment]
```
[1141/1322] Building CXX object modules/gapi/CMakeFiles/opencv_gapi.dir/src/backends/ie/giebackend.cpp.o
FAILED: modules/gapi/CMakeFiles/opencv_gapi.dir/src/backends/ie/giebackend.cpp.o 
$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-c++ -DCVAPI_EXPORTS -DHAVE_DNN_NGRAPH -DHAVE_INF_ENGINE -DHAVE_NGRAPH -DIE_THREAD=IE_THREAD_TBB -DINF_ENGINE_RELEASE=2024000000 -DOPENCV_WITH_ITT=1 -DOV_THREAD=OV_THREAD_TBB -DTBB_PREVIEW_WAITING_FOR_WORKERS=1 -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I$SRC_DIR/build/3rdparty/ippicv/ippicv_lnx/icv/include -I$SRC_DIR/build/3rdparty/ippicv/ippicv_lnx/iw/include -I$SRC_DIR/build -I$SRC_DIR/modules/gapi/include -I$SRC_DIR/build/modules/gapi -I$SRC_DIR/modules/core/include -I$SRC_DIR/modules/flann/include -I$SRC_DIR/modules/imgproc/include -I$SRC_DIR/modules/dnn/include -I$SRC_DIR/modules/features2d/include -I$SRC_DIR/modules/calib3d/include -I$SRC_DIR/modules/video/include -I$SRC_DIR/modules/gapi/src -I$SRC_DIR/modules/gapi/src/3rdparty/vasot/include -I$SRC_DIR/3rdparty/ittnotify/include -I$SRC_DIR/build/3rdparty/ade/ade-0.1.2d/sources/ade/include -isystem $PREFIX/include/eigen3 -fvisibility-inlines-hidden -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/libopencv-4.9.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -D__STDC_CONSTANT_MACROS   -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fopenmp -O3 -DNDEBUG  -DNDEBUG -std=c++17 -fPIC -Wno-error=deprecated-declarations -MD -MT modules/gapi/CMakeFiles/opencv_gapi.dir/src/backends/ie/giebackend.cpp.o -MF modules/gapi/CMakeFiles/opencv_gapi.dir/src/backends/ie/giebackend.cpp.o.d -o modules/gapi/CMakeFiles/opencv_gapi.dir/src/backends/ie/giebackend.cpp.o -c $SRC_DIR/modules/gapi/src/backends/ie/giebackend.cpp
In file included from $SRC_DIR/modules/gapi/src/backends/ie/giebackend.cpp:11:
$SRC_DIR/modules/gapi/src/backends/ie/giebackend.hpp:18:10: fatal error: inference_engine.hpp: No such file or directory
   18 | #include <inference_engine.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[1142/1322] Building CXX object modules/gapi/CMakeFiles/opencv_gapi.dir/src/backends/ocl/goclcore.cpp.o
ninja: build stopped: subcommand failed.

```

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by ilya-lavrenov** [comment]
Good question, for some reason replacement PR https://github.com/conda-forge/opencv-feedstock/pull/397 was also merged

**Comment by hmaarrfk** [comment]
The "why" it was merged was more: "why did you add auto-merge for the bot to merge it automatically".

It was a "nice" way to ask if you forgot to rerender after removing the skips.

#397 was likely merged because github detected similarity???

**Comment by hmaarrfk** [comment]
https://github.com/conda-forge/opencv-feedstock/pull/403 should address the missing builds.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -63,6 +67,8 @@ source:
 
 build:
   number: {{ build }}
+  skip: true  # [py!=311]
```
Comment (h-vetinari) [inline recipe/meta.yaml:70]: Why are we skipping all the other python versions? I understand for development, but this got merged like that?

Comment (h-vetinari) [reply recipe/meta.yaml:70]: Ah sorry, saw that @hmaarrfk already addressed this. I was just looking through the diff and noticed how we're deleting almost all CI configs...

Comment (hmaarrfk) [reply recipe/meta.yaml:70]: right, sorry, we were trouble shooting patches, so i didn't want to hog all the CIs during the debugging process.

50 adds up quickly.

**Pull request merged**